conu/utils/core.py

```diff
@@ -33,68 +35,87 @@ def random_str():
                    for _ in range(random_size))
 
 
-class Volume(object):
-    target = None
-
-    def __init__(self, directory=None, target=None, permissions=None,
-                 facl=None, selinux_type=None, force_selinux=False):
-        self.target = target
-        self.force_selinux = force_selinux
-        if directory:
-            self.directory = directory
-            if not os.path.exists(self.directory):
-                os.makedirs(self.directory)
-        else:
-            self.directory = tempfile.mkdtemp()
-        if permissions:
-            self.set_permission(permissions)
-        if facl:
-            self.set_facl(facl)
-        if selinux_type:
-            self.set_selinux(selinux_type)
-        logger.debug("volume directory: %s" % self.directory)
-
-    def set_target(self, target):
-        self.target = target
-
-    def set_selinux(self, selinux_type):
-        run_cmd(["chcon", "-t", selinux_type, self.directory])
-
-    def set_force_selinux(self, value):
-        self.force_selinux = value
-
-    def set_permission(self, perms):
-        run_cmd(["chmod", perms, self.directory])
-
-    def set_facl(self, rules):
-        for rule in rules:
-            run_cmd(["setfacl", "-m", rule, self.directory])
+class Directory(object):
```

Comment:
I'd suggest moving this to another module and using core(maybe renamed) for little non-class utilities.

Reply:
+1